app/assets/javascripts/components/AccountMenu/Authentication.tsx

```diff
@@ -0,0 +1,382 @@
+import { confirmDialog } from '@Services/alertService';
+import {
+  STRING_ACCOUNT_MENU_UNCHECK_MERGE,
+  STRING_GENERATING_LOGIN_KEYS,
+  STRING_GENERATING_REGISTER_KEYS,
+  STRING_NON_MATCHING_PASSWORDS
+} from '@/strings';
+import { JSXInternal } from '@node_modules/preact/src/jsx';
+import TargetedEvent = JSXInternal.TargetedEvent;
+import TargetedKeyboardEvent = JSXInternal.TargetedKeyboardEvent;
+import { WebApplication } from '@/ui_models/application';
+import { StateUpdater, useEffect, useRef, useState } from 'preact/hooks';
+import TargetedMouseEvent = JSXInternal.TargetedMouseEvent;
+import { User } from '@node_modules/@standardnotes/snjs/dist/@types/services/api/responses';
+import { FC } from 'react';
+
+type Props = {
+  application: WebApplication;
+  server: string | undefined;
+  setServer: StateUpdater<string | undefined>;
+  closeAccountMenu: () => void;
+  notesAndTagsCount: number;
+  showLogin: boolean;
+  setShowLogin: StateUpdater<boolean>;
+  showRegister: boolean;
+  setShowRegister: StateUpdater<boolean>;
+  user: User | undefined;
+}
+
+const Authentication: FC<Props> = ({
+                                     application,
+                                     server,
+                                     setServer,
+                                     closeAccountMenu,
+                                     notesAndTagsCount,
+                                     showLogin,
+                                     setShowLogin,
+                                     showRegister,
+                                     setShowRegister,
+                                     user
+                                   }: Props) => {
+
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [isAuthenticating, setIsAuthenticating] = useState(false);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState<string>('');
+  const [status, setStatus] = useState<string | undefined>(undefined);
+  const [isEmailFocused, setIsEmailFocused] = useState(false);
+
+  const [isEphemeral, setIsEphemeral] = useState(false);
+  const [isStrictSignIn, setIsStrictSignIn] = useState(false);
+  const [shouldMergeLocal, setShouldMergeLocal] = useState(true);
+
+  useEffect(() => {
+    if (isEmailFocused) {
+      emailInputRef.current.focus();
+      setIsEmailFocused(false);
+    }
+  }, [isEmailFocused]);
+
+  // Reset password and confirmation fields when hiding the form
+  useEffect(() => {
+    if (!showLogin && !showRegister) {
+      setPassword('');
+      setPasswordConfirmation('');
+    }
+  }, [showLogin, showRegister]);
+
+  const handleHostInputChange = (event: TargetedEvent<HTMLInputElement>) => {
```

Comment:
Using that type was my first approach actually :) But it doesn't work, throws the following error:

```TS2322: Type '(event: ChangeEvent<HTMLInputElement>) => void' is not assignable to type 'GenericEventHandler<HTMLInputElement>'.
   Types of parameters 'event' and 'event' are incompatible.
     Type 'TargetedEvent<HTMLInputElement, Event>' is missing the following properties from type 'ChangeEvent<HTMLInputElement>': nativeEvent, isDefaultPrevented, isPropagationStopped, persist```

The approach you suggested works in case React is used. But here `GenericEventHandler<HTMLInputElement>` comes from `preact` (we can see it on hovering input's `onChange` in code editor), thus we get that difference.